servers/zms/src/main/java/com/yahoo/athenz/zms/DBService.java

```diff
@@ -7994,6 +8033,117 @@ public DomainList listDomainDependencies(String service) {
         }
     }
 
+    void executeDeleteExpiredMemberships(ResourceContext ctx, ObjectStoreConnection con, int roleId, String roleName,
+                                 int principalId, String normalizedMember, String auditRef, String caller) {
+        try {
+            final String principal = getPrincipalName(ctx);
+
+            // process our delete role member operation
+
+            if (!con.deleteExpiredRoleMember(roleId, principalId, normalizedMember, principal, auditRef)) {
+                throw ZMSUtils.notFoundError(caller + ": unable to delete role member: " +
+                        normalizedMember + " from role: " + roleName, caller);
+            }
+
+            return;
+
+        } catch (ResourceException ex) {
```

Comment:
why are we catching and throwing the exception. We should just remove the try / cache block.